.github/workflows/security-scan.yml

```
@@ -0,0 +1,218 @@
+
+    - name: Setup dependencies
+      run: |
+        npm ci
+
+    - name: Run static security scan
+      run: |
+        npm audit -f markdown | tee npm-audit-report.md >> $GITHUB_STEP_SUMMARY
+        npm audit --json > npm-audit-report.json || true
+
+    - name: Upload npm audit report
+      uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)